stats/bookkeeper-stats-api/src/main/java/org/apache/bookkeeper/stats/ThreadRegistry.java

```diff
@@ -40,14 +51,16 @@ public static void register(String threadPool, int threadPoolThread) {
      */
     public static void register(String threadPool, int threadPoolThread, long threadId) {
         ThreadPoolThread tpt = new ThreadPoolThread(threadPool, threadPoolThread, threadId);
-        threadPoolMap.put(threadId, tpt);
+        ThreadPoolThread previous = threadPoolMap.put(threadId, tpt);
+        assert previous == null : "Thread " + threadId + " was already registered in thread pool " + previous.threadPool + " as thread " + previous.ordinal;
```

Comment:
I am not sure that assertions are enabled in BK, maybe an old style "if (....) throw new IllegalStateException is better here

Reply:
Sure, I'll update that now.

Reply:
@eolivelli assert is not encouraged by ivankelly #995